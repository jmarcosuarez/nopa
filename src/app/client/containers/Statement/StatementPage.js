import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router';
import * as actions from '../../redux/actions';
import * as transactionsSelectors from '../../redux/reducers/bank';

import styles from './Statement.css';
import { Layout, Button, AccDetails, ListView, ListRow } from '../../components';

class StatementPage extends Component {
  componentDidMount() {
    // this will be here only for testing purposes
    this.props.onFetchTransactions();
  }
  renderLoading() {
    return (
      <p>Loading...</p>
    );
  }

  renderRow(transactionId, transaction) {
    return (
      <ListRow
        rowId={transactionId}
        // onClick={this.onRowClick}
      >
        <h3>{transaction.beneficary}</h3>
        <p>{transaction.value}</p>
      </ListRow>
    );
  }

  render() {
    if (!this.props.transactionsById) return this.renderLoading();
    return (
      <Layout title="Statements Page!">
        <div className={styles.main}>
          <h1>Which bank does this account belong to?</h1>
          <p>Track all of your payments by connecting as many bank accounts as you&#39;d<br />
              like to your Nopa account and get updates on your balance instantly. Plus it&#39;s free.</p>   

          <AccDetails />

          <ListView
            rowsIdArray={this.props.transactionsIdArray}
            rowsById={this.props.transactionsById}
            renderRow={this.renderRow} />

          <div className={styles.button}>
            <Button>Show More</Button>
          </div>

        </div>
      </Layout>
    );
  }
} 

StatementPage.propTypes = {
  onFetchTransactions: PropTypes.func.isRequired,
  transactionsById: PropTypes.array,
  transactionsIdArray: PropTypes.array,
};

StatementPage.defaultProps = {
  transactionsById: [],
  transactionsIdArray: [],
};

function mapStateToProps(state) {
  const [transactionsById, transactionsIdArray] = transactionsSelectors.getTransactions(state);
  return {
    transactionsById,
    transactionsIdArray,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onFetchTransactions: bindActionCreators(actions.fetchTransactions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(StatementPage);
