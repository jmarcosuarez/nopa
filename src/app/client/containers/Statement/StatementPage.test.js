/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import StatementPage from './StatementPage';

describe('Renders correctly', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);

  // Initialize mockstore with empty state
  const initialState = {
    user: 
    {
      surname: 'Denis', 
      sortCode: '23-43-54',
      accountNumber: '09434534343434',
      passCode: '0943',
      memorableWord: 'Qwteyruy',
      loggedIn: false,
    },
    bank: {
      choosenBank: "HCBC",
      transactions: [
        { id: 1, dateStr: "Now", beneficary: "PAYPAL ZARA", value: "- Â£35.98" },
      ],
    },
  };
  const store = mockStore(initialState);

  it('should render correctly', () => {
    const output = shallow(
      <StatementPage store={store} />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
