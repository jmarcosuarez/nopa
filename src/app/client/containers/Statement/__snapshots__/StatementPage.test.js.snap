// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<section>
  <div
    className="header"
  >
    <div
      className="logo"
    />
    <div
      className="signin"
    >
      <div
        className=""
      >
        <a
          className="button"
          onClick={[Function]}
          style={Object {}}
        >
          Log In
        </a>
      </div>
    </div>
  </div>
  <div
    className="layout"
  />
  <section
    className="partners"
  >
    <div>
      <span>
        Our partners:
      </span>
    </div>
    <div>
      <div>
        <img
          alt="Airbnb"
          className="Nopa"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="Metro"
          className="Nopa"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="Pariti"
          className="Nopa"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="Unshackled"
          className="Nopa"
          src="test-file-stub"
        />
      </div>
    </div>
  </section>
  <footer
    className="footer"
  >
    <div>
      <p>
        © Nopa Limited 2017 All rights reserved. ‘Nopa’ and the Nopa logo are trade marks of Nopa Limited. Nopa is a member of CIFAS - the UK's leading anti-fraud association, and we are registered with the Office of the Information Commissioner (No. Z879078).
      </p>
      <p>
        Nopa Limited is incorporated in England & Wales (registration number 05197592), with its registered office at 1st Floor, Cottons Centre, Tooley Street, London, SE1 2QG. Nopa Limited is authorised and regulated by the Financial Conduct Authority, and entered on the Financial Services Register under firm registration number 563134.
      </p>
    </div>
  </footer>
</section>
`;

exports[`Renders correctly should render correctly 1`] = `
<StatementPage
  bank={
    Object {
      "choosenBank": "HCBC",
      "transactions": Array [
        Object {
          "beneficary": "PAYPAL ZARA",
          "dateStr": "Now",
          "id": 1,
          "value": "- £35.98",
        },
      ],
    }
  }
  onFetchTransactions={[Function]}
  onLogInUser={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  transactionsById={
    Array [
      Object {
        "beneficary": "PAYPAL ZARA",
        "dateStr": "Now",
        "id": 1,
        "value": "- £35.98",
      },
    ]
  }
  transactionsIdArray={
    Array [
      "0",
    ]
  }
  user={
    Object {
      "accountNumber": "09434534343434",
      "loggedIn": false,
      "memorableWord": "Qwteyruy",
      "passCode": "0943",
      "sortCode": "23-43-54",
      "surname": "Denis",
    }
  }
/>
`;
